pyfr/plugins/fluidforce.py

```diff
@@ -120,7 +158,30 @@ def __call__(self, intg):
             norms = self._norms[etype, fidx]
 
             # Do the quadrature
-            f += np.einsum('i...,ij,jik', qwts, p, norms)
+            f[:ndims] += np.einsum('i...,ij,jik', qwts, p, norms)
+
+            if self._viscous:
+                # Get operator and J^-T matrix
+                m4 = self._m4[etype]
+                rcpjact = self._rcpjact[etype, fidx]
+
+                # Transformed gradient at solution points
+                tduupts = np.dot(m4, uupts.reshape(nupts, -1))
+                tduupts = tduupts.reshape(ndims, nupts, nvars, -1)
+
+                # Physical gradient at solution points
+                duupts = np.einsum('ijkl,jkml->ikml', rcpjact, tduupts)
+                duupts = duupts.reshape(ndims, nupts, -1)
+
+                # Interpolate gradient to flux points
+                dufpts = np.array([np.dot(m0, duupts[i]) for i in range(ndims)])
+                dufpts = dufpts.reshape(ndims, nfpts, nvars, -1).swapaxes(1, 2)
+
+                # Viscous stress
+                vis = self.stress_tensor(ufpts, dufpts)
+
+                # Do the quadrature
+                f[ndims:] += np.einsum('i,klij,jil', qwts, vis, norms)
```

Comment:
One final comment before we merge.  We should probably be consistent about if we do `i,` or `i...,` for `einsum` when we do quadrature.  Either is fine.